tests/database/test_db_diff.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+
+from evm.db.diff import (
+    DBDiff,
+    DBDiffTracker,
+    DiffMissingError,
+)
+
+
+@pytest.fixture
+def db():
+    return DBDiffTracker()
+
+
+def test_database_api_get(db):
+    db[b'key-1'] = b'value-1'
+
+    assert db.get(b'key-1') == b'value-1'
+    assert db[b'key-1'] == b'value-1'
+
+
+def test_database_api_set(db):
+    db[b'key-1'] = b'value-1'
+    assert db[b'key-1'] == b'value-1'
+    db[b'key-1'] = b'value-2'
+    assert db[b'key-1'] == b'value-2'
+
+    db[b'key-1'] = b'value-1'
+    assert db[b'key-1'] == b'value-1'
+    db[b'key-1'] = b'value-2'
+    assert db[b'key-1'] == b'value-2'
+
+
+def test_database_api_existence_checking(db):
+    assert b'key-1' not in db
+
+    db[b'key-1'] = b'value-1'
+
+    assert b'key-1' in db
+
+
+def test_database_api_delete(db):
+    db[b'key-1'] = b'value-1'
+    db[b'key-2'] = b'value-2'
+
+    assert b'key-1' in db
+    assert b'key-2' in db
+
+    del db[b'key-1']
+    del db[b'key-2']
+
+    assert b'key-1' not in db
+    assert b'key-2' not in db
+
+
+def test_database_api_missing_key_retrieval(db):
+    assert db.get(b'does-not-exist') is None
+
+    try:
+        val = db[b'does-not-exist']
+    except DiffMissingError as exc:
+        assert not exc.is_deleted()
```

Comment:
I was surprised that this was a function and not a property.  I think functions named like this are foot guns because were this line to be `if exc.is_deleted` it would always evaluate to `True` because `bool(fn)` returns `True`.